specs/language/resources.tex

```diff
@@ -4,170 +4,118 @@
-
-\Sub{Dimensions}{Resources.rotybufs.dims}
-
-The size of a Buffer can be retrieved using the \texttt{GetDimensions} method.
-\begin{HLSL}
-void GetDimensions(out uint size);
-\end{HLSL}
-
-This returns the full size of the buffer in bytes through the \texttt{width} out parameter.
-
-\Sub{Element Access}{Resources.rotybufs.access}
-
-The contents of Buffers can be retrieved using the \texttt{Load} methods
-or the subscript operator.
-
-\begin{HLSL}
- T Load(in int pos);
- T operator[](uint pos);
-\end{HLSL}
-
-These each return the element at the given position \texttt{pos} of the type specified in the buffer definition.
 
-An additional \texttt{Load} method returns the indicated element just as the other,
-but also takes a second \texttt{status} parameter that returns information about the accessed resource.
-\begin{HLSL}
- T Load(in int pos, out uint status);
-\end{HLSL}
-
-The \texttt{status} parameter returns an indication of whether the value retrieved accessed a mapped tile
-resource on tile-based platforms. This parameter is interpretted using the \texttt{CheckAccessFullyMapped}
-built-in function. If tiled resources are enabled, it returns true if the value was accessed from a mapped
-tile resource and false otherwise. If tiled resourced are not enabled, it will return true.
-
-\Sec{Writable Typed Buffers}{Resources.rwtybufs}
 
-\begin{HLSL}
 template <typename T>
- class RWBuffer {
-
+ class RWBuffer : public Buffer {
+ public:
    RWBuffer();
+   RWBuffer(RWBuffer buf);
    RWBuffer operator=(RWBuffer buf);
 
-   void GetDimensions(out uint size);
-
    // element access
-   T operator[](uint n);
-   T Load(in int pos);
-   T Load(in int pos, out uint status);
+   T &operator[](uint n);
 };
+}
 \end{HLSL}
 
-\Sub{Constructors}{Resources.rwtybufs.ctrs}
+\Sub{Constructors}{Resources.tybufs.ctrs}
+
+Read-only Buffers can be defined with explicit or implicit template parameter types.
+\begin{HLSL}
+  Buffer buf1;
+  Buffer<> buf2;
+  Buffer<float4> buf3;
+\end{HLSL}
+Since \texttt{float4} is the default type, all of these definitions are equivalent.
 
 RWBuffers must be defined with explicit template parameter types.
 \begin{HLSL}
   RWBuffer<float4> buf;
 \end{HLSL}
 
-When defined at global scope, RWBuffers are bound to externally-defined backing stores
+When defined at global scope, typed buffers are bound to externally-defined backing stores
 using the explicit binding location if provided or the implicit binding if not.
 
-When defined at local scope, RWBuffers represent local references
+When defined at local scope, typed buffers represent local references
 that can be associated with global buffers when assigned,
 but must be resolvable to a unique global buffer.
 
 \begin{HLSL}
-  RWBuffer<int> gbuf;
-  void doit() {
-    RWBuffer lbuf<int> = gbuf;
+  Buffer<int> grobuf;
+  RWBuffer<int> grwbuf;
+  void main() {
+    Buffer lrobuf<int> = grobuf;
+    RWBuffer lrwbuf<int> = grwbuf;
```

Comment:
```suggestion
    Buffer<int> lrobuf = grobuf;
    RWBuffer<int> lrwbuf = grwbuf;
```

Reply:
I fixed this location and one other where I'd somehow gotten it out of order.